cmd/kueuectl/app/list/helpers.go

```diff
@@ -49,3 +85,31 @@ func addActiveFilterFlagVar(cmd *cobra.Command, p *[]bool) {
 	cmd.Flags().BoolSliceVar(p, "active", make([]bool, 0),
 		"Filter by active status. Valid values: 'true' and 'false'.")
 }
+
+func addForObjectFilterFlagVar(cmd *cobra.Command, p *string) {
+	cmd.Flags().StringVar(p, "for", "",
+		"Filter pods to only those pertaining to the specified resource types.")
+}
+
+// parseForObjectFilterFlag parses the user input --for flag and returns a objectRef struct
+func parseForObjectFilterFlag(input string) (objectRef, error) {
+	objRef := objectRef{}
+	if input == "" {
+		return objRef, nil
+	}
+
+	parts := strings.Split(input, "/")
+	if len(parts) > 2 {
+		return objRef, fmt.Errorf("invalid value '%s' used in --for flag; value must be in the format [TYPE[.API-GROUP]/]NAME", input)
+	}
+
+	if len(parts) == 1 {
+		objRef.Name = parts[0]
+		return objRef, nil
+	}
+
+	objRef.Name = parts[1]
+	objRef.Kind, objRef.APIGroup, _ = strings.Cut(parts[0], ".")
```

Comment:
Could you please check this:

https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/events/events.go#L171

And this:

https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/events/events.go#L378-L410

Is it useful in our case?

Reply:
IIUC the using GVK instead objectRef.

Reply:
[Here](https://github.com/kubernetes-sigs/kueue/pull/2238#discussion_r1622275041) discussion about this topic. Just to don't missed.

Reply:
Thanks for that. Should I just need to re-use the same [decodeResourceTypeName](https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/events/events.go#L380-L410)? Looks like it expects `RESTMapper` as well.

Reply:
I think yes. Yes, we have `RESTMapper ` on `clientGetter`.

Reply:
Yeah I used that. For `testing/fake.go` I did created a reciever for `WithRESTMapper` to create the mapper. But it looks like we need to add every `gvk`'s in the unit tests?

Something like this? Not sure if this the right way to use them in tests. Let me know if I'm wrong, thanks!
```go
mapperGVKs := []schema.GroupVersionKind{
    batchv1.SchemeGroupVersion.WithKind("Job"), # Similarly for all GVKs kubeflows...?
}

gvs := make([]schema.GroupVersion, len(mapperGVKs))
for _, gvk := range mapperGVKs {
    fmt.Println(gvk)
    gvs = append(gvs, gvk.GroupVersion())
}

mapper := meta.NewDefaultRESTMapper(gvs)
for _, gvk := range mapperGVKs {
    mapper.Add(gvk, meta.RESTScopeNamespace)
}

tf := cmdtesting.NewTestClientGetter()
tf.WithNamespace(metav1.NamespaceDefault)
tf.WithRestMapper(mapper)
```

Reply:
Exactly, on each test cases, where we need it, we can add slice of `GroupVersionKind`. 
Not sure that we need to add gvs. Try like this `mapper := meta.NewDefaultRESTMapper(nil)`.

Reply:
Not sure that we need to add gvs. Try like this `mapper := meta.NewDefaultRESTMapper(nil)`.

Reply:
Yeah it'll error our during mapping step [here](https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/events/events.go#L393-L401). So I think we need to pass all required GVKs for tests.
`Error: no matches for /, Resource=job`

Reply:
Yes, sure. GVKs need to add.